### PR TITLE
removed redundant CreateMembershipIfNeeded call

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
A potential race was being caused by a redundant call to CreateMembershipIfNeeded .

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-33396

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by streamer45 — APPROVED** [review]
Nice :+1:

**Review by agnivade — CHANGES_REQUESTED** [review]
@reflog - Sorry for not having any prior context on this, but can you please explain what exactly is the race here in the PR message?

Can we also add a test for this?

**Comment by streamer45** [comment]
> @reflog - Sorry for not having any prior context on this, but can you please explain what exactly is the race here in the PR message?

@agnivade 

This was spotted during load-tests. The logs were getting filled with "duplicate entries" warnings on inserting `ThreadMembership` rows. It turned out there were two subsequent calls (one of which in a separate goroutine) triggered by `handlePostEvents`. That's where the race originated from. Since we are making db calls it didn't have any real side-effect other than causing a spam of logs (and possibly a few more queries).

**Comment by reflog** [comment]
> 
> 
> > @reflog - Sorry for not having any prior context on this, but can you please explain what exactly is the race here in the PR message?
> 
> @agnivade
> 
> This was spotted during load-tests. The logs were getting filled with "duplicate entries" warnings on inserting `ThreadMembership` rows. It turned out there were two subsequent calls (one of which in a separate goroutine) triggered by `handlePostEvents`. That's where the race originated from. Since we are making db calls it didn't have any real side-effect other than causing a spam of logs (and possibly a few more queries).
☝🏻 
What Claudio said :)

File: app/post.go [hunk 1]
```diff
@@ -458,12 +458,6 @@ func (a *App) handlePostEvents(post *model.Post, user *model.User, channel *mode
 		return err
 	}
```
Comment (agnivade) [inline app/post.go:460]: It seems like there is some difference in the two. This query wouldn't be executed if `ThreadAutoFollow` is set to false and if it's a root post. Whereas the other one will be executed always, along with the `ThreadAutoFollow` passed to it. 

I guess the end result is the same, but I am wondering if we should add the if condition to whichever query we are keeping?

**Comment by reflog** [comment]
@agnivade @streamer45 - I am closing this PR in favor of https://github.com/mattermost/mattermost-server/pull/17068 since both Agniva's comment was a correct one and there were other minor changes.

**Pull request closed without merging**